server/vcs/provider/github/pr_commit_fetcher.go

```diff
@@ -0,0 +1,46 @@
+package github
+
+import (
+	"context"
+	gh "github.com/google/go-github/v45/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"time"
+)
+
+type PRCommitFetcher struct {
+	ClientCreator githubapp.ClientCreator
+}
+
+func (c *PRCommitFetcher) FetchLatestCommitTime(ctx context.Context, installationToken int64, repo models.Repo, prNum int) (time.Time, error) {
+	client, err := c.ClientCreator.NewInstallationClient(installationToken)
+	if err != nil {
+		return time.Time{}, errors.Wrap(err, "creating installation client")
+	}
+	run := func(ctx context.Context, nextPage int) ([]*gh.RepositoryCommit, *gh.Response, error) {
+		listOptions := gh.ListOptions{
+			PerPage: 100,
+		}
+		listOptions.Page = nextPage
+		return client.PullRequests.ListCommits(ctx, repo.Owner, repo.Name, prNum, &listOptions)
+	}
+	commits, err := Iterate(ctx, run)
+	if err != nil {
+		return time.Time{}, errors.Wrap(err, "iterating through entries")
+	}
+	latestCommitTimestamp := time.Time{}
+	for _, commit := range commits {
+		if commit.GetCommit() == nil {
+			return time.Time{}, errors.New("getting latest commit")
+		}
+		if commit.GetCommit().GetCommitter() == nil {
+			return time.Time{}, errors.New("getting latest committer")
+		}
```

Comment:
Technically api returns pointers to these objects, so I figured with a pointer comes the risk of being nil. I don't expect them to be nil, but in case GH does have issues sending a full response, I think there is no harm in writing these nil checks to be safe.

Reply:
Ok, and is returning an error necessary here.  Again, would that impact the UX? Is there a failover mechanism we can apply here to minimize impact to the customer?

Reply:
For any error, the UX would see that the check run still fails. The original failed policy check result would be outputted to the check run details, but we would prepend a "internal server error" message to the top of it.